repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,266 @@
+                    "You should either add it to your config file, "
+                    "or use the download commang without --config"
+                )
+            trusted_root = decode_trusted_root(trusted_root)
+
+    if hash_prefix:
+        artifacts_url = update_artifact_url_with_hash_prefix(artifacts_url)
+
+    ok = download_artifact(
+        metadata_url,
+        artifacts_url,
+        artifact_name,
+        trusted_root,
+        directory_prefix,
+    )
+
+    if ok:
+        console.print(
+            f"Successfully completed artifact download: {artifact_name}"
+        )
+
+
+def update_artifact_url_with_hash_prefix(artifact_url: str) -> str:
+    artifact_file_name = artifact_url.split("/")[-1]
+    hash_prefix = find_hash_bin(artifact_file_name)
+    new_artifact_file_name = hash_prefix + "-" + artifact_file_name
+    return artifact_url.rsplit("/", 1)[0] + "/" + new_artifact_file_name
+
+
+def decode_trusted_root(trusted_root: base64) -> str:
+    trusted_root_encoded = base64.b64encode(bytes(trusted_root, "utf-8"))
+    return trusted_root_encoded.decode("utf-8")
+
+
+def build_metadata_dir(metadata_url: str) -> str:
+    """build a unique and reproducible directory name for the repository url"""
+    name = sha256(metadata_url.encode()).hexdigest()[:8]
+    return f"{Path.home()}/.local/share/rstuf/{name}"
+
+
+def init_tofu(metadata_url: str, root_url: Optional[str]) -> bool:
+    """Initialize local trusted metadata (Trust-On-First-Use) and create a
+    directory for downloads"""
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    if not os.path.isdir(metadata_dir):
+        os.makedirs(metadata_dir)
+
+    if not root_url:
+        root_url = f"{metadata_url}/metadata/1.root.json"
+    try:
+        request.urlretrieve(root_url, f"{metadata_dir}/root.json")
+    except OSError:
+        console.print(f"Failed to download initial root from {root_url}")
+        return False
+
+    console.print(
+        f"Trust-on-First-Use: Initialized new root in {metadata_dir}"
+    )
+    return True
+
+
+def download_artifact(
+    metadata_url: str,
+    artifacts_url: str,
+    artifact_name: str,
+    trusted_root: Optional[str],
+    directory_prefix: Optional[str],
+) -> bool:
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    if not os.path.isfile(f"{metadata_dir}/root.json"):
+        console.print(
+            "Trusted local root not found. Using 'tofu' to "
+            "Trust-On-First-Use or copy trusted root metadata to "
+            f"{metadata_dir}/root.json"
+        )
+        ok = init_tofu(metadata_url, trusted_root)
+        if not ok:
+            console.print(
+                f"Trusted local root not found in {metadata_url}"
+                "`tofu` was not successful"
+            )
+            return False
+
+    print(f"Using trusted root in {metadata_dir}")
+
+    if directory_prefix:
+        download_dir = f"{directory_prefix}/downloads"
```

Comment:
We shouldn't create a dir, the user decides the prefix
```suggestion
        download_dir = f"{directory_prefix}/"
```